core/server/master/src/main/java/alluxio/master/file/meta/LazyUfsBlockLocationCache.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.file.meta;
+
+import alluxio.AlluxioURI;
+import alluxio.Configuration;
+import alluxio.PropertyKey;
+import alluxio.exception.InvalidPathException;
+import alluxio.underfs.UnderFileSystem;
+import alluxio.underfs.options.FileLocationOptions;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.List;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Lazily cache the block locations only when needed.
+ */
+@ThreadSafe
+public class LazyUfsBlockLocationCache implements UfsBlockLocationCache {
+  private static final Logger LOG = LoggerFactory.getLogger(LazyUfsBlockLocationCache.class);
+
+  /** Number of blocks to cache. */
+  private static final int MAX_BLOCKS =
```

Comment:
Does `0` really disable this? Is a 0-sized cache the same as doing no caching?

Reply:
the cached item is immediately evicted, so there is overhead.